plugins/scaffolder-backend/src/scaffolder/actions/builtin/bitbucket-cloud/bitbucketCloudRepoCreate.ts

```diff
@@ -0,0 +1,193 @@
+          },
+          token: {
+            title: 'Authentication Token',
+            type: 'string',
+            description:
+              'The token to use for authorization to BitBucket Cloud',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            title: 'A URL to the repository with the provider',
+            type: 'string',
+          },
+          repoContentsUrl: {
+            title: 'A URL to the root of the repository',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        description,
+        defaultBranch = 'master',
+        enablePipeline = false,
+        repoVisibility = 'private',
+        token,
+        repoUrl,
+      } = ctx.input;
+
+      const {
+        apiBaseUrl,
+        authorization,
+        integrationConfig,
+        project,
+        repo,
+        workspace,
+      } = getBitbucketCloudConfig({ repoUrl, token, integrations });
+
+      // Use Bitbucket API to create a new repository
+      const { remoteUrl, repoContentsUrl } =
+        await bitbucketCloudCreateRepository({
+          authorization,
+          workspace: workspace || '',
+          project,
+          repo,
+          repoVisibility,
+          mainBranch: defaultBranch,
+          description,
+          apiBaseUrl,
+        });
+
+      const gitAuthorInfo = {
+        name: config.getOptionalString('scaffolder.defaultAuthor.name'),
+        email: config.getOptionalString('scaffolder.defaultAuthor.email'),
+      };
+
+      let auth;
+
+      if (ctx.input.token) {
+        auth = {
+          username: 'x-token-auth',
+          password: ctx.input.token,
+        };
+      } else {
+        if (
+          !integrationConfig.config.username ||
+          !integrationConfig.config.appPassword
+        ) {
+          throw new Error(
+            'Credentials for Bitbucket Cloud integration required for this action.',
+          );
+        }
+
+        auth = {
+          username: integrationConfig.config.username,
+          password: integrationConfig.config.appPassword,
+        };
+      }
+      // Use isomorphic-git to push workspace to new repository
+      await initRepoAndPush({
+        dir: getRepoSourceDirectory(ctx.workspacePath, ctx.input.sourcePath),
+        remoteUrl,
+        auth,
+        defaultBranch,
+        logger: ctx.logger,
+        commitMessage: config.getOptionalString(
+          'scaffolder.defaultCommitMessage',
+        ),
+        gitAuthorInfo,
+      });
+      // Use Bitbucket API to enable pipeline
+      if (enablePipeline) {
+        await bitbucketCloudEnablePipeline({
+          workspace,
+          repo,
+          authorization,
+          apiBaseUrl,
+        });
+      }
+
+      ctx.output('remoteUrl', remoteUrl);
+      ctx.output('repoContentsUrl', repoContentsUrl);
+      ctx.output('enablePipeline', enablePipeline);
```

Comment:
This isn't in the output schema - should it be here?

Reply:
I have removed this line in the next commit. Since the function does not return a value, just passing the input through to the output felt messy.

Thanks for your review!